pkgs/development/interpreters/octave/build-octave-package.nix

```diff
@@ -82,63 +89,64 @@ let
     "passthru"
   ];
 
-in
-stdenv.mkDerivation (
-  {
-    packageName = "${fullLibName}";
-    # The name of the octave package ends up being
-    # "octave-version-package-version"
-    name = "${octave.pname}-${octave.version}-${fullLibName}";
+  self = stdenv.mkDerivation (
```

Comment:
If we are just wrapping `stdenv.mkDerivation`, can't we just use `finalAttrs`? And in the `testOctaveBuildEnv` attribute use `finalAttrs.finalPackage`?

Reply:
from op:

> While the `self` approach here works, this builder ought to be refactored to use `lib.extendMkDerivation` instead, but I leave that for the octave maintainers.

Reply:
But I can switch to `finalAttrs` instead of `self` :+1: